From 4b3bf882f44bb05874bfcb61f9eb59d96116099f Mon Sep 17 00:00:00 2001
From: Jookia <166291@gmail.com>
Date: Wed, 28 Aug 2019 21:48:31 +1000
Subject: [PATCH 17/17] thermal: imx: enable hwmon explicitly

hwmon is not enabled by the thermal driver by default.
xobs fixed this by adding a special 'hwmon' property to the Novena
device tree, but never mainlined.
The Raspberry Pi hit this same issue and after working with upstream
came to the solution in 6a7c0215d28c6ec1482d3aa5bf28d88c2070f426,
which is to have the specific driver enable hwmon.
Do the same thing here.
---
 drivers/thermal/imx_thermal.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/drivers/thermal/imx_thermal.c b/drivers/thermal/imx_thermal.c
index aa452acb60b6..d9fcf75908ce 100644
--- a/drivers/thermal/imx_thermal.c
+++ b/drivers/thermal/imx_thermal.c
@@ -23,6 +23,8 @@
 #include <linux/types.h>
 #include <linux/nvmem-consumer.h>
 
+#include "../thermal_hwmon.h"
+
 #define REG_SET		0x4
 #define REG_CLR		0x8
 #define REG_TOG		0xc
@@ -798,6 +800,17 @@ static int imx_thermal_probe(struct platform_device *pdev)
 		return ret;
 	}
 
+	data->tz->tzp->no_hwmon = false;
+	ret = thermal_add_hwmon_sysfs(data->tz);
+	if (ret < 0) {
+		dev_err(&pdev->dev, "failed to add hwmon interface: %d\n", ret);
+		clk_disable_unprepare(data->thermal_clk);
+		thermal_zone_device_unregister(data->tz);
+		cpufreq_cooling_unregister(data->cdev);
+		cpufreq_cpu_put(data->policy);
+		return ret;
+	}
+
 	dev_info(&pdev->dev, "%s CPU temperature grade - max:%dC"
 		 " critical:%dC passive:%dC\n", data->temp_grade,
 		 data->temp_max / 1000, data->temp_critical / 1000,
-- 
2.20.1

